src/main/java/courseManager/Course.java

```diff
@@ -0,0 +1,109 @@
+package courseManager;
+
+import java.util.ArrayList;
+
+import java.time.DayOfWeek;
+import java.time.LocalTime;
+
+
+public class Course {
+    private String courseCode;
+    private String courseName;
+    private String semester;
+    private boolean archived;
+    private float credit;
+
+    private Outline outline;
+
+    private ArrayList<CourseEvent> courseEvents;
+
+    /**
+     * Create a new Course object with the given parameters
+     * @param courseCode the code of the course
+     *                   e.g. 'CSC207'
+     * @param courseName the name of the course
+     *                   e.g. 'Software Design'
+     * @param semester the semester the course is offered in
+     *                   e.g. 'Fall 2022'
+     * @param credit the number of credits the course is worth; must be either 0.5 or 1.0
+     * @param archived whether the course is archived or not
+     */
+
+    public Course(String courseCode, String courseName, String semester, boolean archived, float credit) {
+        this.courseCode = courseCode;
+        this.courseName = courseName;
+        this.semester = semester;
+        this.archived = archived;
+        this.credit = credit;
+        this.courseEvents = new ArrayList<>();
+    }
+
+    public void setOutline(Outline outline) {
```

Comment:
I think we discussed this in previous lab, but basically there's no need for a setOutline function since and outline cannot exist outside of the context of a course. So to fix this, just have the Course constructor create an empty outline for itself.